### ISSUE TITLE
[IMPROVE] Consistence with Unity indentation

### ISSUE THREAD

**Issue by Simferoce** [mackysoft/Unity-SerializeReferenceExtensions#57]
### Improvement destription

The indentation of the field does not correspond to Unity expected indentation for nested SerializeReference.

Indentation between the foldout and the label:
![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/12441beb-3049-44f7-af67-4b68a4121ca5)
![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/1275c85b-7d5d-414f-b973-74a7cd404cf3)

The position of the foldout element does not correspond to what is expected either. Note: unity use 15 indent pixel.
![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/4758da09-6207-40c9-979c-0d0240a57783)

![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/c0a73bd2-30b0-4be4-b67a-f85794c30eec)

The dropdown is more indented then what we expect.
![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/5f5ffb3f-69ad-4c4d-a5ca-0481e0fd6140)
![image](https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/23125629/2c8b0190-88de-438f-adad-a0cb6d20540a)

Schema:
(AbilityDefinition - ScriptableObject)
-- [SerializeReference] (Ability - Plain Old C# Object)
---- [SerializeReference] (Statistic - IStatisticFloat)

Version: 2022.1.8f1, Library Version: 1.3.1

**Comment by Maligan** [comment]
Oh, I'm digging into the problem and realize that Unity has some regression here...
Have a fix which solves mentioned but only on version 2020.
2022 & 2024 have weird behavior

Tried to add dirty hack for those versions but stuck into the problem:
Nested foldouts aren't drawn properly even with custom drawer which only use`EditorGUI.PropertyField()` and nothing else

For example this for comparing versions:

2020 (all is correct)
<img width="509" alt="2020 - correct" src="https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/1263469/a63caf44-e401-4e26-8463-cafaf74f61d1">

2022 (look at Property 2/3 arrow and label horizontal positions)
<img width="566" alt="2022 - incorrect" src="https://github.com/mackysoft/Unity-SerializeReferenceExtensions/assets/1263469/f6dacf2f-f32b-4225-8125-f3d0f05d7528">

Unity Tech added some bugs into `EditorGUI.Foldout` control, but I can't understand where, and I don't have an Idea how to fix it - any nested property foldouts drawn incorrect if you use custom drawer.

**Comment by mackysoft** [comment]
In the last update, I used the define directive to address the drawing issue between Unity versions, but it didn't completely fix it.

I also tried to migrate to UIElements, but the current UnityEditor API is insufficient here to completely replace the functionality of this package from IMGUI to UIElements.
https://github.com/mackysoft/Unity-SerializeReferenceExtensions/issues/51#issuecomment-2266849134

I'll close this issue once as it's not critical.
